plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,521 @@
+
+    def __init__(self, module):
+        super().__init__(module)
+        self.result["response"] = []
+        self.supported_states = ["merged"]
+        self.payload = module.params
+        # Need to follow Camel Case to snake case
+        # So we are using the Key maping for API fields
+        self.keymap = {
+            "led_brightness_level": "ledBrightnessLevel",
+            "led_status": "ledStatus",
+            "primary_ip_address": "primaryIpAddress",
+            "eth_mac": "ethMac",
+            "ap_name": "apName",
+            "ap_ethernet_mac_address": "apEthernetMacAddress",
+            "ap_manager_interface_ip": "apManagerInterfaceIp",
+            "associated_wlc_ip": "associatedWlcIp",
+            "boot_date_time": "bootDateTime",
+            "collection_interval": "collectionInterval",
+            "collection_status": "collectionStatus",
+            "description": "description",
+            "device_support_level": "deviceSupportLevel",
+            "error_code": "errorCode",
+            "error_description": "errorDescription",
+            "family": "family",
+            "hostname": "hostname",
+            "id": "id",
+            "instance_tenantId": "instanceTenantId",
+            "instance_uuid": "instanceUuid",
+            "interface_count": "interfaceCount",
+            "inventory_status_detail": "inventoryStatusDetail",
+            "last_update_time": "lastUpdateTime",
+            "last_updated": "lastUpdated",
+            "line_card_count": "lineCardCount",
+            "line_card_id": "lineCardId",
+            "location": "location",
+            "location_name": "locationName",
+            "mac_address": "macAddress",
+            "managed_atleast_once": "managedAtleastOnce",
+            "management_ip_address": "managementIpAddress",
+            "management_state": "managementState",
+            "memory_size": "memorySize",
+            "platform_id": "platformId",
+            "reachability_failure_reason": "reachabilityFailureReason",
+            "reachability_status": "reachabilityStatus",
+            "role": "role",
+            "role_source": "roleSource",
+            "serial_number": "serialNumber",
+            "series": "series",
+            "snmp_contact": "snmpContact",
+            "snmp_location": "snmpLocation",
+            "software_type": "softwareType",
+            "software_version": "softwareVersion",
+            "tag_count": "tagCount",
+            "tunnel_udp_port": "tunnelUdpPort",
+            "type": "type",
+            "up_time": "upTime",
+            "uptime_seconds": "uptimeSeconds",
+            "waas_device_mode": "waasDeviceMode"
+        }
+
+        self.baseurl = "https://" + module.params["dnac_host"]+ ":" + module.params["dnac_port"]
+        self.log('Login DNAC using by user: ' + module.params["dnac_username"], "INFO")
+        try:
+            self.dnac = api.DNACenterAPI(base_url=self.baseurl, 
```

Comment:
Not need...setting up the connection will be part of main function only.

Reply:
We are using self.dnac on multiple place. hence if it is there in class will be better